receiver/awsfirehosereceiver/internal/unmarshaler/cwlog/unmarshaler.go

```diff
@@ -70,22 +70,40 @@ func (u *Unmarshaler) UnmarshalLogs(compressedRecord []byte) (plog.Logs, error)
 	byResource := make(map[resourceKey]plog.LogRecordSlice)
 
 	// Multiple logs in each record separated by newline character
+	var anyErrors bool
 	scanner := bufio.NewScanner(r)
 	for datumIndex := 0; scanner.Scan(); datumIndex++ {
 		var log cWLog
 		if err := jsoniter.ConfigFastest.Unmarshal(scanner.Bytes(), &log); err != nil {
+			anyErrors = true
 			u.logger.Error(
 				"Unable to unmarshal input",
 				zap.Error(err),
 				zap.Int("datum_index", datumIndex),
 			)
 			continue
 		}
-		if !isValid(log) {
-			u.logger.Error(
-				"Invalid log",
-				zap.Int("datum_index", datumIndex),
-			)
+
+		switch log.MessageType {
```

Comment:
I personally found this function flow hard to follow, and I'm wondering if this block could be extracted into some kind of validation function, so it would be easier to read, and wouldn't need those `continue` statements.

Reply:
I've extracted some of the logic into a new `parseLog` function. It's still necessary to have some `continue` statements, but they're less prolific now.